llvm/test/ThinLTO/X86/call-calloc.ll

```diff
@@ -0,0 +1,35 @@
+; RUN: opt -module-summary %s -o %t1.bc
+; RUN: opt -module-summary %p/Inputs/calloc.ll -o %t2.bc
+; RUN: llvm-lto2 run -save-temps -o %t.out \
+; RUN:   %t1.bc -r=%t1.bc,_Z7DoStuffv,px -r=%t1.bc,_ZnwmRKSt9nothrow_t,px \
+; RUN:          -r=%t1.bc,_ZSt7nothrow -r=%t1.bc,malloc,l \
+; RUN:   %t2.bc -r=%t2.bc,malloc,pl -r=%t2.bc,calloc,pl
+; RUN: llvm-dis %t.out.2.5.precodegen.bc -o - | FileCheck %s
+
+; CHECK: define {{.*}} @calloc
```

Comment:
Actually, since you are only checking that the calloc def is not removed, can this test be simplified even further? I.e. you really only need a single source file with a def of calloc run through llvm-lto2, do you even need any of the other code?

Reply:
Honestly, I'm not quite sure if such simplification is justified. May be it is better to also check that malloc/memset pair is folded to calloc, so we have both def and use?

Reply:
Sure that seems fine too. My point was just that if we aren't checking anything from that main file, then there is no need to have it, as I don't believe it affects whether calloc is kept in the other file. And if the operator new isn't needed, but it is sufficient to just have a function that directly calls both malloc and memset in order to get the folding to calloc, then just have that and remove anything extraneous.

Reply:
We want to test references from one TU to both `malloc` and `calloc` with a stub implementation of `calloc`. I think it is more genuine for `calloc.ll` to define `malloc` along with `calloc` (otherwise `malloc` is undefined while `calloc` is defined, making the test less like a real world scenario).